dotnet-authserver/src/TeacherIdentity.AuthServer/Journeys/CreateUserHelper.cs

```diff
@@ -84,6 +84,12 @@ public async Task<User> CreateUserWithTrn(AuthenticationState authenticationStat
 
         _dbContext.Users.Add(user);
 
+        if (authenticationState.TrnToken is not null && authenticationState.TrnAssociationSource == TrnAssociationSource.TrnToken)
+        {
+            _dbContext.Database.ExecuteSqlInterpolatedAsync(
+                $"update trn_tokens set user_id = {userId} where trn_token = {authenticationState.TrnToken};");
```

Comment:
Adding TrnToken back to AuthenticationState means that the new TrnAssociationSource property is somewhat redundant. It can be calculated at this point based on TrnToken != null, etc. Any thoughts on keeping it?

Reply:
If it's redundant I'd say we get rid of it.